src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -809,19 +842,9 @@ internal static Action<Diagnostic> GetDiagnosticSinkWithSuppression(Action<Diagn
             };
         }
 
-        private static Diagnostic GetFilteredDiagnostic(Diagnostic diagnostic, Compilation compilation, ISymbol symbolOpt = null)
+        private static Diagnostic GetFilteredDiagnostic(Diagnostic diagnostic, Compilation compilation)
         {
-            var filteredDiagnostic = compilation.Options.FilterDiagnostic(diagnostic);
```

Comment:
This code looks familiar -- I think this change will give a merge conflict with your "emit suppressed issues" work.